src/mbgl/tile/tile_cache.cpp

```diff
@@ -8,7 +8,7 @@ void TileCache::setSize(size_t size_) {
 
     while (orderedKeys.size() > size) {
         auto key = orderedKeys.front();
-        orderedKeys.pop_front();
+        orderedKeys.remove(key);
```

Comment:
why is this change?

Reply:
In the case of reducing the size of the cache, a situation could turn out when the tile was no longer there and the mention of it in the orderedKeys remained, which led to the triggering of the assert.
This was possible due to a second error in the if-condition with emplace.